client/src/containers/PostDashboard.tsx

```diff
@@ -11,46 +11,68 @@ import Header from '../components/Header';
 import PostPreview from '../components/PostPreview';
 import CreatePost from '../components/CreatePost';
 
-import ServerApi, { APIResponse, PostUser } from '../api/v1';
+import ServerApi, { PostUserPreview } from '../api/v1';
 import { useNavigate } from 'react-router-dom';
 
+const POSTS_PER_PAGE = 25; // Maximum (previewable) posts per page
+
 const api = new ServerApi();
 
 export const UserContext: React.Context<{
   isLoading?: boolean;
   data?: User | null;
 }> = React.createContext({});
 
-function RecentPosts() {
-  const [recentPosts, updateRecent] = React.useState([] as PostUser[]);
-
-  const fetchRecentPosts = (limit: number, offset: number) => {
-    api
-      .fetchRecentPosts(limit, offset)
-      .then(
-        (
-          res: APIResponse<{ data: { result?: PostUser[]; count: number } }>
-        ) => {
-          if (res.data && res.data.data.result) {
-            updateRecent(res.data.data.result);
-          }
-        }
-      )
-      .catch((err) => console.log(err));
-  };
+function RecentPosts(props: {
+  setPageCount: React.Dispatch<React.SetStateAction<number>>;
+  pageNum: number;
+}) {
+  const [recentPosts, updateRecent] = React.useState([] as PostUserPreview[]);
+  const [openedPost, setOpenedPost] = React.useState(false);
 
+  /* Fetch new posts by polling */
   React.useEffect(() => {
     const interval = setInterval(() => {
-      fetchRecentPosts(15, 0);
+      if (!openedPost) {
+        api
+          .fetchRecentPosts(
+            POSTS_PER_PAGE,
+            POSTS_PER_PAGE * (props.pageNum - 1)
+          )
+          .then((res) => {
+            if (res.data && res.data.data.result) {
+              updateRecent(res.data.data.result);
+              props.setPageCount(
+                Math.ceil(res.data.data.count / POSTS_PER_PAGE)
+              );
+            }
+          })
+          .catch((err) => console.log(err));
+      }
     }, 500);
 
     return () => clearInterval(interval);
   });
 
+  /* Fetch posts triggered by page-change or post dialog close */
+  React.useEffect(() => {
+    if (!openedPost) {
+      api
+        .fetchRecentPosts(POSTS_PER_PAGE, POSTS_PER_PAGE * (props.pageNum - 1))
+        .then((res) => {
+          if (res.data && res.data.data.result) {
+            updateRecent(res.data.data.result);
+            props.setPageCount(Math.ceil(res.data.data.count / POSTS_PER_PAGE));
```

Comment:
I would write some tests for the pagination logic to make sure the correct page is being displayed and the correct data is being displayed. You can just create some mock data to be returned from the mock `fetchRecentPosts` method and just assert that the UI displays correctly.